core/auth/authorizer.js

```diff
@@ -59,51 +49,51 @@ const canEditRecord = (user, record) => {
 
   const recordDataStep = Record.getStep(record)
 
-  const userAuthGroup = User.getAuthGroups(user).find(authGroup => AuthGroups.getSurveyUuid(authGroup) === Record.getSurveyUuid(record))
+  const userAuthGroup = User.getAuthGroupBySurveyUuid(Record.getSurveyUuid(record))(user)
 
   // level = 'all' or 'own'. If 'own', user can only edit the records that he created
   // If 'all', he can edit all survey's records
-  const level = R.path([keys.recordSteps, recordDataStep], userAuthGroup)
+  const level = AuthGroup.getRecordEditLevel(recordDataStep)(userAuthGroup)
 
   return level === keys.all || (level === keys.own && Record.getOwnerUuid(record) === User.getUuid(user))
 }
 
 const canEditRecordsInDataQuery = R.pipe(
-  getSurveyUserGroup,
+  _getSurveyUserGroup,
   authGroup => R.includes(
-    AuthGroups.getName(authGroup),
+    AuthGroup.getName(authGroup),
     [
-      AuthGroups.groupNames.systemAdmin,
-      AuthGroups.groupNames.surveyAdmin,
-      AuthGroups.groupNames.surveyEditor,
-      AuthGroups.groupNames.dataAnalyst,
-      AuthGroups.groupNames.dataCleanser
+      AuthGroup.groupNames.systemAdmin,
+      AuthGroup.groupNames.surveyAdmin,
+      AuthGroup.groupNames.surveyEditor,
+      AuthGroup.groupNames.dataAnalyst,
+      AuthGroup.groupNames.dataCleanser
     ]
   )
 )
 
-const canAnalyzeRecords = hasSurveyPermission(permissions.recordAnalyse)
+const canAnalyzeRecords = _hasSurveyPermission(permissions.recordAnalyse)
 
 // ======
 // ====== Users
 // ======
 
 // CREATE
-const canInviteUsers = hasSurveyPermission(permissions.userInvite)
+const canInviteUsers = _hasSurveyPermission(permissions.userInvite)
 
 // READ
 const canViewUser = (user, surveyInfo, userToView) => {
   return User.isSystemAdmin(user) || (
-    !!getSurveyUserGroup(user, surveyInfo, false) &&
-    !!getSurveyUserGroup(userToView, surveyInfo, false)
+    !!_getSurveyUserGroup(user, surveyInfo, false) &&
```

Comment:
Are the `!!` necessary here? It just coerces the value to boolean?
Is it because these functions are defined to return booleans rather than the "equivalent" true-y/falsy values?